packages/db/scripts/migrations/migrate-quizzes-to-v2.ts

```diff
@@ -0,0 +1,130 @@
+#!/usr/bin/env tsx
+
+/**
+ * Quiz V1 to V2 Migration Script
+ *
+ * Purpose: Migrate legacy quiz format (V1) to new discriminated union format (V2)
+ *
+ * V1 Format: Array of multiple-choice questions
+ * V2 Format: Object with version field and questions array supporting multiple types
+ *
+ * Safety: Run with --dry-run first to preview changes
+ */
+import { upgradeQuizzes } from "../../../aila/src/protocol/schemas/quiz/conversion/lessonPlanQuizMigrator";
+import { aiLogger } from "@oakai/logger";
+
+import { PrismaClient } from "@prisma/client";
+
+const log = aiLogger("db");
+
+function hasQuizzes(output: unknown): boolean {
+  if (!output || typeof output !== "object") return false;
+
+  const data = output as {
+    lessonPlan?: { starterQuiz?: unknown; exitQuiz?: unknown };
+  };
+  return Boolean(data.lessonPlan?.starterQuiz || data.lessonPlan?.exitQuiz);
+}
+
+async function main() {
+  const isDryRun = process.argv.includes("--dry-run");
+  const includeDeleted = process.argv.includes("--include-deleted");
+  const showHelp = process.argv.includes("--help");
+
+  if (showHelp) {
+    log.info(`
+Quiz V1 to V2 Migration
+
+Usage: pnpm tsx migrate-quizzes-to-v2.ts [--dry-run] [--include-deleted]
+
+Options:
+  --dry-run          Preview changes without modifying database
+  --include-deleted  Also migrate soft-deleted sessions
+  --help             Show this help
+`);
+    process.exit(0);
+  }
+
+  const prisma = new PrismaClient();
+
+  log.info(`Starting quiz migration (${isDryRun ? "DRY RUN" : "LIVE"})`);
+  if (includeDeleted) {
+    log.info("Including deleted sessions");
+  }
+
+  try {
+    // Find all sessions with quizzes
+    // Note: We'll check for quiz presence in the code since Prisma JSON filtering is limited
+    const allSessions = await prisma.appSession.findMany({
+      where: includeDeleted ? {} : { deletedAt: null },
```

Comment:
I guess we'd always want to include deleted?

Reply:
I wasn't sure so I made it an option, but we can commit in a direction. I think I agree with you, but I wasn't sure if a deleted item was an archive of a point in time. I'll make it update deleted ones all the time